### PR TITLE
make a non-escalating policy resource group

### PR DESCRIPTION
**Pull request by deads2k**
Scope down cluster-reader.  I guess I forgot to open this yesterday.

@liggitt @smarterclayton

### CODE REVIEW

File: pkg/authorization/api/types.go [hunk 1]
```diff
@@ -71,6 +77,29 @@ var (
 		KubeExposedGroupName:   {"pods", "replicationcontrollers", "serviceaccounts", "services", "endpoints", "persistentvolumeclaims", "pods/log"},
 		KubeAllGroupName:       {KubeInternalsGroupName, KubeExposedGroupName, QuotaGroupName},
 		KubeStatusGroupName:    {"pods/status", "resourcequotas/status", "namespaces/status"},
+
+		OpenshiftNonEscalatingViewableGroupName: {
```
Comment (smarterclayton) [inline pkg/authorization/api/types.go]: In the future you might want to have a test case that negates the set and verifies nothing changes (i.e verify with code that NonEscalating contains nothing from escalating

Comment (liggitt) [inline pkg/authorization/api/types.go]: yeah, there is no chance we catch drift here without a test

Comment (smarterclayton) [inline pkg/authorization/api/types.go]: Even an init func that panics is fine.  Would rather panic early

Comment (smarterclayton) [inline pkg/authorization/api/types.go]: Let's add that now `func init() { util.NewStringSet(KubeNonEscalatingViewableGroupName).Diff() }`

**Comment by deads2k** [comment]
updated.

**Comment by smarterclayton** [comment]
LGTM [merge]

**Comment by smarterclayton** [comment]
```
- FAIL: TestSubjects (0.01s)
	authorizer_test.go:400: groups: Expected [RootUsers system:cluster-admins system:cluster-readers system:nodes], got [RootUsers system:cluster-admins system:nodes]
FAIL
coverage: 73.7% of statements
```

File: pkg/authorization/api/types.go [hunk 2]
```diff
@@ -71,9 +82,21 @@ var (
 		KubeExposedGroupName:   {"pods", "replicationcontrollers", "serviceaccounts", "services", "endpoints", "persistentvolumeclaims", "pods/log"},
 		KubeAllGroupName:       {KubeInternalsGroupName, KubeExposedGroupName, QuotaGroupName},
 		KubeStatusGroupName:    {"pods/status", "resourcequotas/status", "namespaces/status"},
+
+		OpenshiftEscalatingViewableGroupName: {"oauthauthorizetokens", "oauthaccesstokens"},
+		KubeEscalatingViewableGroupName:      {"secrets"},
+		NonEscalatingResourcesGroupName:      {OpenshiftEscalatingViewableGroupName, KubeEscalatingViewableGroupName},
```
Comment (liggitt) [inline pkg/authorization/api/types.go]: these names are backwards, right?

Comment (liggitt) [inline pkg/authorization/api/types.go]: Yup:
```
$ oc policy who-can get secrets
Users:  system:openshift-client

Groups: system:cluster-admins
        system:cluster-readers
        system:nodes

$ oc policy who-can get pods
Users:  system:openshift-client
        system:serviceaccount:default:deployer
        system:serviceaccount:openshift-infra:build-controller
        system:serviceaccount:openshift-infra:deployment-controller

Groups: system:cluster-admins
        system:nodes
```
Worth a sanity check test?

**Comment by deads2k** [comment]
Tests did fail, but I added an other very simple sanity test and a unit test to check coverage of cluster-readers.  re[merge]

**Comment by deads2k** [comment]
gofmt

**Pull request merged**